oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/token/OAuth2TokenIssuer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.server.authorization.token;
+
+import org.springframework.security.oauth2.core.AbstractOAuth2Token;
+
+/**
+ * Base interface for token issuer implementations. Implementations can chose to return any type of an OAuth2 token,
+ * {@code TOKEN_TYPE} which implements {@code AbstractOAuth2Token}.
+ *
+ * @see JwtAccessTokenOAuth2TokenIssuer
+ * @see OpaqueRefreshTokenOAuth2TokenIssuer
+ *
+ * @author Alexey Nesterov
+ * @since 0.0.4
+ */
+public interface OAuth2TokenIssuer<TOKEN_TYPE extends AbstractOAuth2Token> {
```

Comment:
Please make a copy of `OAuth2Token` in this repository, similar to this [commit](https://github.com/spring-projects-experimental/spring-authorization-server/commit/bf24cfb19e85ee2dd20701e20baed1bbed23fec0#diff-24bbe651f1e6b03c25ad8cf4853602cdf2422ceb94a2762659ee1b3d5889bdf4R22) and use this for the type instead of `AbstractOAuth2Token`. We cannot do GA releases that are based on milestone dependencies so I had to downgrade Spring Security to 5.4.+.

Reply:
I'll have to copy `OAuth2AccessToken` as well and make both refresh and access token implement this copied `OAuth2Token` interface. I guess it's fine for now, all of these classes will be removed after migration to 5.5.